design-proposals/testing-and-ci/properly-using-context-in-functional-tests.md

```diff
@@ -0,0 +1,133 @@
+# Properly use contexts in functional tests
+
+## Overview
+This proposal is about properly use golang contexts in KubeVirt functional tests.
+
+## Introduction
+Since the start of the KubeVirt project, contexts became a very important concept
+in golang, when writing an asynchronous logic. In a way, the KubeVirt code left
+behind regarding the proper usage of golang contexts.
+
+Using `context.Background()` or even worse, `context.TODO()` is a code smell, because
```

Comment:
Could not just be the termination of the application but also termination of a goroutine.